File: apps/admin-v2/src/domain/repositories/notification.repository.ts
import type { AppNotification } from '../entities/app-notification';

export interface NotificationRepository {
  getAll(unreadOnly?: boolean): Promise<{ notifications: AppNotification[]; unreadCount: number }>;
  markAsRead(id: string): Promise<void>;
  markAllAsRead(): Promise<void>;
  registerDeviceToken(token: string, platform: string): Promise<void>;
  removeDeviceToken(token: string): Promise<void>;
}

File: apps/customer_v2/lib/features/notifications/domain/repositories/notification_repository.dart
// lib/features/notifications/domain/repositories/notification_repository.dart
import 'package:fpdart/fpdart.dart';
import '../../../../core/error/failures.dart';
import '../entities/app_notification.dart';

abstract class NotificationRepository {
  Future<Either<Failure, List<AppNotification>>> getAll({bool unreadOnly});
  Future<Either<Failure, int>> getUnreadCount();
  Future<Either<Failure, Unit>> markAsRead(String id);
  Future<Either<Failure, Unit>> markAllAsRead();
}
